### PR TITLE
complete orderService

### PR DESCRIPTION
**Pull request by ngchwanlii**
added code to OrderServiceImpl.java

TODO: still need to fix orderMapper bug

### CODE REVIEW

File: account/src/main/java/com/caldevsupplychain/account/service/AccountServiceImpl.java [hunk 1]
```diff
@@ -1,19 +1,5 @@
 package com.caldevsupplychain.account.service;
 
-import java.util.List;
```
Comment (calvin-liang) [inline account/src/main/java/com/caldevsupplychain/account/service/AccountServiceImpl.java]: You intellij is reordering imports in a diff order. Can you change the setting to keep it consistent?
screen shot 2017-11-12 at 10 28 20 am
screen shot 2017-11-12 at 10 28 46 am
screen shot 2017-11-12 at 10 28 57 am

File: order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	private OrderService orderService;
+	private AccountService accountService;
+
+	private OrderValidator orderValidator;
+
+	private OrderMapper orderMapper;
+
+	private ApiErrorsExceptionHandler apiErrorsExceptionHandler;
+
+	@PostMapping("/order")
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java:46]: don't need `/order`. the endpoint `/api/v1/orders` for create is good.

File: order/src/main/java/com/caldevsupplychain/order/vo/OrderWS.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+@Getter
+@Setter
+public class OrderWS implements Serializable {
+
+	private static final long serialVersionUID = -8744630767752497964L;
+
+	private String uuid;
+	private String displayId;
+	private Long id;
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/vo/OrderWS.java:18]: remove `private Long id;`. `uuid` is the id we expose to public.
use `uuid` as `userId` and `agentId`. maybe name them userUuid to make it clear

File: order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:create")
+	public ResponseEntity<?> createOrder(@Validated @RequestBody OrderWS orderWS) {
+
+		// check user_id
+		if(!accountService.findById(orderWS.getUserId()).isPresent()){
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java:52]: use `findById` here. we are exposing uuid to the public, not id(usually want to hide underneath implementations to the public. In this case it's the fact that we are using primary key from DB)

File: order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+			log.error("Error in create order. User id={} not found", orderWS.getUserId().toString());
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.USER_ID_NOT_FOUND.name(), "In create order user id={}" + orderWS.getUserId().toString().toString() + " not found"), HttpStatus.NOT_FOUND);
+		}
+
+		// check agent_id
+		if(!accountService.findById(orderWS.getAgentId()).isPresent()){
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java:58]: find by uuid, you also need to verify the user role here. also error if user is agent id is not pointing to an agent.

File: order/src/main/java/com/caldevsupplychain/order/model/Order.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	@Column(name = "order_note")
+	private String orderNote;
+
+	@PrePersist
+	@Override
+	protected void onCreate() {
+		super.onCreate();
+		uuid = UUID.randomUUID().toString();
+		String[] arr = uuid.split("-");
+		displayId = arr[arr.length-1].toUpperCase();
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/model/Order.java:69]: 👍

File: order/src/main/java/com/caldevsupplychain/order/service/OrderService.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.List;
+import java.util.Optional;
+
+public interface OrderService {
+
+	OrderBean createOrder(OrderBean orderBean);
+
+	OrderBean updateOrder(OrderBean orderBean);
+
+	OrderBean deleteOrder(OrderBean orderBean);
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/service/OrderService.java:16]: You can use id param for delete `OrderBean deleteOrder(Long orderId);`

File: order/src/main/java/com/caldevsupplychain/order/service/OrderServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+		if(order != null) {
+			return Optional.of(orderMapper.toBean(order, new CycleAvoidingMappingContext()));
+		}
+		return Optional.empty();
+	}
+
+	@Override
+	public Optional<List<OrderBean>> findByDisplayId(String displayId) {
+		List<Order> orders = orderRepository.findByDisplayId(displayId, new PageRequest(0, Integer.MAX_VALUE));
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/service/OrderServiceImpl.java:87]: getting `MAX_VALUE` orders(all orders) will be dangerous when the table is big.

File: order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java (hunk 2 above)
Comment (ngchwanlii) [reply order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java:46]: fixed

File: order/src/main/java/com/caldevsupplychain/order/vo/OrderBean.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.List;
+
+@Getter
+@Setter
+public class OrderBean {
+
+	private String uuid;
+	private String displayId;
+	private Long id;
+	private Long userId;
```
Comment (calvin-liang) [inline order/src/main/java/com/caldevsupplychain/order/vo/OrderBean.java:17]: `userUuid, agentUuid`

File: order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java (hunk 4 above)
Comment (ngchwanlii) [reply order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java:52]: do you mean findByUuid?

File: order/src/main/java/com/caldevsupplychain/order/vo/OrderWS.java (hunk 3 above)
Comment (ngchwanlii) [reply order/src/main/java/com/caldevsupplychain/order/vo/OrderWS.java:18]: Fixed

File: order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java (hunk 4 above)
Comment (ngchwanlii) [reply order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java:52]: Fixed

File: order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java (hunk 5 above)
Comment (ngchwanlii) [reply order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java:58]: Fixed

File: order/src/main/java/com/caldevsupplychain/order/service/OrderServiceImpl.java (hunk 8 above)
Comment (ngchwanlii) [reply order/src/main/java/com/caldevsupplychain/order/service/OrderServiceImpl.java:87]: Yea it depends on how much front end wanted to show the orders per pagination. I can reset this afterward, to 10 or something

File: order/src/main/java/com/caldevsupplychain/order/vo/OrderBean.java (hunk 9 above)
Comment (ngchwanlii) [reply order/src/main/java/com/caldevsupplychain/order/vo/OrderBean.java:17]: fixed

**Pull request merged**